tests/programs/dlopen_demo.sh

```diff
@@ -6,17 +6,31 @@
 # Copyright The Mbed TLS Contributors
 # SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
 
-. "${0%/*}/../../scripts/demo_common.sh"
+DEMO_COMMON_NEED_QUERY_COMPILE_TIME_CONFIG=0
+
+SCRIPT_DIR=$(dirname "$0")
+. "${SCRIPT_DIR}/../../scripts/project_detection.sh"
+. "${SCRIPT_DIR}/../../scripts/demo_common.sh"
 
 msg "Test the dynamic loading of libmbed*"
 
-program="$programs_dir/test/dlopen"
-library_dir="$root_dir/library"
+# Once demo_common.sh is sourced we're granted to be either in Mbed TLS or
+# TF-PSA-Crypto root folders and IS_MBEDTLS_ROOT variable can be used to
+# discriminate between the two.
```

Comment:
IS_MBEDTLS_ROOT has been removed and this is a bit misleading I'd say. We are granted to be in Mbed TLS or TF-PSA-Crypto tree but not to be at the root of the tree (if I am not missing something) thus we should keep using the `programs_dir` and `root_dir` variables, not `pwd`.